src/snowflake/cli/plugins/nativeapp/run_processor.py

```diff
@@ -65,102 +68,69 @@
-            show_app_row = self.get_existing_app_info()
+    @classmethod
+    def versioned_dev(cls, version: str, patch: Optional[int] = None):
+        return cls(False, version, patch)
 
-            # 3. If existing application object is found, perform a few validations and upgrade the application object.
-            if show_app_row:
+    @classmethod
+    def release_directive(cls):
+        return cls(False, from_release_directive=True)
 
-                # Check if not created by Snowflake CLI or not created using "files on a named stage" / stage dev mode.
-                if show_app_row[COMMENT_COL] not in ALLOWED_SPECIAL_COMMENTS or (
-                    show_app_row[VERSION_COL] != LOOSE_FILES_MAGIC_VERSION
-                ):
-                    raise ApplicationAlreadyExistsError(self.app_name)
+    @property
+    def is_dev_mode(self) -> bool:
+        return not self._from_release_directive
 
-                # Check for the right owner
-                ensure_correct_owner(
-                    row=show_app_row, role=self.app_role, obj_name=self.app_name
-                )
+    def using_clause(self, app: NativeAppProjectModel) -> str:
+        if self._from_release_directive:
+            return ""
 
-                # If all the above checks are in order, proceed to upgrade
-                try:
-                    cc.step(f"Upgrading existing application object {self.app_name}.")
-                    self._execute_query(
-                        f"alter application {self.app_name} upgrade using @{self.stage_fqn}"
-                    )
+        if self.version:
+            patch_clause = f"patch {self.patch}" if self.patch else ""
+            return f"using version {self.version} {patch_clause}"
 
-                    # if debug_mode is present (controlled), ensure it is up-to-date
-                    if self.debug_mode is not None:
-                        self._execute_query(
-                            f"alter application {self.app_name} set debug_mode = {self.debug_mode}"
-                        )
+        stage_name = StageManager.quote_stage_name(app.stage_fqn)
+        return f"using {stage_name}"
 
-                    self._execute_post_deploy_hooks()
-                    return
+    def ensure_app_usable(self, app: NativeAppProjectModel, show_app_row: dict):
+        """Raise an exception if we cannot proceed with install given the pre-existing application object"""
 
-                except ProgrammingError as err:
-                    if err.errno not in UPGRADE_RESTRICTION_CODES:
-                        generic_sql_error_handler(err)
-                    else:
-                        cc.warning(err.msg)
-                        self.drop_application_before_upgrade(policy, is_interactive)
+        if self._requires_created_by_cli:
+            if show_app_row[COMMENT_COL] not in ALLOWED_SPECIAL_COMMENTS or (
+                show_app_row[VERSION_COL] != LOOSE_FILES_MAGIC_VERSION
```

Comment:
My hot take: I don't think we need this `VERSION_COL` check anymore. The error handler that handles cross-installation-method errors will capture this later and allow the user to action on it, rather than having to just seen an exception here and bailing out

Reply:
Sounds good, get rid of it.

Reply:
Will be addressed in a subsequent PR.